source-git-monorepos/README.md

```diff
@@ -50,21 +89,367 @@ packages:
+repositories found in `--dist-git-store`. The command would figure out
+dist-git subdirectory names in `--dist-git-store`, based on
+`downstream_package_name`s found in `.distro/source-git.yaml`. If any of them
+is missing, the command would fail early.
+
+```
+SYNOPSIS
+       packit source-git update-dist-git [OPTIONS] --dist-git-store PATH SOURCE_GIT
+```
+
+Example:
+
+```shell
+$ packit source-git update-dist-git --dist-git-store rpms/ src/copr
+```
+
+### `packit source-git update-source-git`
+
+Let's not change the interface of this command and don't support the usecase
+to update a single source-git monorepo from multiple dist-git repositories in
+order to avoid complications. Expect that multiple commands are going to be
+executed if updates need to be taken from multiple dist-git repositories.
+
+On the other hand, the command needs to be updated to support a source-git
+repo as a target when the source-git repo is a monorepo. In this case only the
+paths of the sub-package need to be considered during the update. `DIST_GIT`
+should match one of the `downstream_package_name`s in `.distro/source-git.yaml`.
+
+### `packit source-git status`
+
+The command should check the sync status of a sub-package from a source-git
+monorepo and the corresponding dist-git repo. `downstream_package_name` should
+match `basename DIST_GIT`.
+
+### Hardly
+
+When opening an MR in a source-git monorepo, Hardly should figure out which
+dist-git packages are going to be impacted by the change (by looking at
+`package_paths`), and open mirror-MRs in the dist-git repositories of those
+packages.
+
+CI results from all mirror-MRs should be synced back to the source-git MR.
+
+**Open question**: Is the behavior above good enough? Can there be source-git
+changes happening outside the package paths, but which influence the dist-git
+content of the package?
+
+## Impact on Packit
+
+All things that follow assume that [the `metadata` key was dropped].
+
+Additionally to what was presented above, a new property of job-objects in
+`.packit.yaml` needs to be introduced, to be used to specify which packages
+the job should handle when triggered. The property would be called `packages`
+and would be a list of package-object names. If a job-object has no such key,
+all packages of a monorepo should be handled.
+
+Example:
+
+```yaml
+upstream_project_url: https://pagure.io/copr/copr.git
+packages:
+  copr-backend:
+    specfile_path: backend/copr-backend.spec
+    package_paths:
+      - backend
+  copr-frontend:
+    specfile_path: frontend/copr-frontend.spec
+    package_paths:
+      - frontend
+  copr-cli:
+    specfile_path: cli/copr-cli.spec
+    package_paths:
+      - cli
+jobs:
+  - job: copr_build
+    trigger: pull_request
+    targets:
+      - fedora-all
+      - epel-8
+    packages:
+      - copr-frontend
+      - copr-backend
+  - job: copr_build
+    trigger: pull_request
+    targets:
+      - fedora-all
+    packages:
+      - copr-cli
```

Comment:
Obviously this is just and example for how the configuration should look like, which by chance it uses `copr`, but in no way should this be considered as a reference config for `copr`.

Reply:
Since nobody except for us will use/read this example, I think it is fine.  But if you wanted to use this as an example in docs, I would prefer switching it (clients supported everywhere, servers only on Fedora, as more (and newer) deps are typically needed).